ethcore/src/state_db.rs

```diff
@@ -125,14 +172,107 @@ impl StateDB {
  			try!(Self::commit_bloom(batch, bloom_lock.drain_journal()));
  		}
 		let records = try!(self.db.commit(batch, now, id, end));
-		if self.is_canon {
-			self.commit_cache();
-		} else {
-			self.clear_cache();
-		}
+		self.commit_hash = Some(id.clone());
+		self.commit_number = Some(now);
 		Ok(records)
 	}
 
+	/// Apply buffered cache changes and synchronize canonical
```

Comment:
"buffered cache changes" what does this mean? "cache change" here presumably doesn't mean "population form disk"? "Apply" _to what_?